packages/proto-build/src/code_generator.rs

```diff
@@ -124,22 +126,39 @@ impl CodeGenerator {
         let deps_dirs = self
             .deps
             .iter()
-            .map(|dep| self.root.join(&dep.project_dir))
+            .map(|dep| (self.root.join(&dep.project_dir), dep.clone()))
```

Comment:
can we somehow have private struct or name tuple value so that we can distinguish what they are instead of `.0`/ `.1` ?